lib/engine/game/g_18_eus/step/buy_sell_par_shares.rb

```diff
@@ -76,6 +78,20 @@ def modify_purchase_price(bundle)
 
             super
           end
+
+          def process_buy_shares(action)
+            return super unless action.bundle.corporation == @game.bny
+
+            @round.bought_from_ipo = true if action.bundle.owner.corporation?
+            buy_shares(action.entity, action.bundle, swap: action.swap, allow_president_change: false)
+            track_action(action, action.bundle.corporation)
+            log_pass(action.entity)
```

Comment:
This is there because in the base process_buy_shares, when buy_shares is called, it doesn't pass in allow_president_change which defaults to "true".  We want to that to be "false" for the bny.  

I suppose I could add allow_president_change to the base buy_shares action - didn't know if that was right approach, either.

Reply:
Could also add a call out from process_buy_shares to @game.allow_president_change?(corporation) ?

Reply:
Added a call allow_president_change? to the step (instead of to @game).